Serializer registration improvements (small)

Joshua-Ferguson opened the pull request:
These improvements are aimed at helping users better diagnose serializer mistakes.
(sorry for empty PR before this)

dnfclas commented:
Hi __@Joshua-Ferguson__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla2.dotnetfoundation.org.
    </span>

TTYL, DNFBOT;

dnfclas commented:
__@Joshua-Ferguson__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, DNFBOT;

jdom commented on src/Orleans/AssemblyLoader/AssemblyProcessor.cs:
```diff
@@ -145,17 +145,24 @@ private static void ProcessAssembly(Assembly assembly)
+                catch (Exception exception)
                 {
-                    SerializationManager.FindSerializationInfo(type);
+                    Logger.Error(ErrorCode.SerMgr_TypeRegistrationFailure, $"Failed to load type {type.FullName} in assembly {assembly.FullName}.", exception);
```
please don't use c# 6 features yet, as this will break some of our CI environments. We intend to migrate soon, but in the meantime this would completely stall our functional tests and load/reliability tests.

jdom commented on src/Orleans/AssemblyLoader/AssemblyProcessor.cs:
```diff
@@ -145,17 +145,24 @@ private static void ProcessAssembly(Assembly assembly)
+                    GrainFactory.FindSupportClasses(type);
                 }
-                if (shouldProcessSerialization)
+                catch (Exception exception)
```
I assume it was a conscious decision to now log an error but continue processing other types and start the silo, even after those errors occurred. Correct?

jdom commented:
Thanks @Joshua-Ferguson. Other than not using c# 6, the change looks good. Could you also squash into 1 commit?

Joshua-Ferguson commented:
I removed the C#6 syntax and squashed it (first squash).

Yes, continuing the loop was a on purpose. My idea is, if it aborts and doesn't fully register the assembly types, it might mask the serializer error. There is no downstream try/catch so rethrowing the exception won't do anything, other than break the loop.

veikkoeeva commented:
@Joshua-Ferguson It looks like you succeeded well with your first squash. :) Just in case, I find [this handy](http://stackoverflow.com/questions/12528854/how-to-perform-rebase-squash-using-tortoisegit) when I'm in the mood of fiddling with a text editor.

jdom commented:
@Joshua-Ferguson one additional thing, can you rebase it on top of the current master? If you can't, I can just pull the changes locally and do it myself. You'll still show as the author of the commit, but the PR will have to be closed manually so github won't show it as "Merged" in the the UI.

Joshua-Ferguson commented:
I've had the _quite the experience_ today, trying to rebase on top of the current master.  I'm not sure if I exceeded, except for that now it says, 'This branch has no conflicts with the base branch' instead of the there is conflicts notification.  It seems 'git rebase master' is super dangerous.

Did I succeed? How do I know?

jdom commented:
Well, kind of. For some reason there's 1 merge commit merging 2 old parents that already exist in master, not sure how it got into that state (and not sure if it's harmful or not, other than the "dirty" commit history).
I strongly suggest that when you do rebases, that you use a GUI tool. In particular I use TortoiseGit, but others work well too (such as SourceTree).

In theory what you would have to do at this point to fix it, is another interactive rebase skipping that unnecessary merge commit. 

Having said that, I don't want to take more of your time (unless you actually want to learn the process regardless of this particular PR). So again I offer to do the rebase locally if that's what you prefer.

Joshua-Ferguson commented:
I have broken it. I could not get it to rebase to master. I got this pretty graph with all these dots, but I can't make the dots connect they way I want.

sebastianburckhardt commented:
I have had trouble with using "git rebase" to squash commits before, I think it does not always work right if your version history is slightly non-standard (i.e. more complicated than the simple diagrams always shown in blogs on this topic). 

For what it's worth: I now always squash my commits manually, without using any of git's history-altering functionality like "git rebase". It's easy if you have two repositories on the same machine. I create a fresh branch in one repo, then manually copy over the files from the other repo, with the branch I want to squash. If the original branch is not important to me, I throw it away locally, and replace it with the new one (force-push for remote repositories). If the original branch is important to me (say I have dependent branches that I want to keep working with), I merge the squashed branch into it - that way, I can keep both the squashed and the non-squashed branch around, and keep  working with dependent branches.

jdom commented:
Hi @Joshua-Ferguson, I'm sorry you had a bad experience with Git. Is there a way I can help you with this? Do you still have a branch with the code in your local machine? If so, we can still salvage your fix, which we very much want and appreciate. I can help you go over it and recover it. If you'd like, we could even do some screen sharing so I can show you how to do an interactive rebase.

Joshua-Ferguson commented:
Thanks for the kind words. Its not anyone else's fault.
I think the main issue I had was how I updated my fork to be current. There was a button I saw on the github webpage (I went back to find it, but I can't) and when I clicked it, it let me issue a PR to my fork of all the commits I didn't have. In all the directions I read online, didn't ever have that step. 

At one point I managed to issue an anti-PR to that PR and then I sqaushed them. It looked okay, except the head transformed to a commit from January. And Then when I tried to fix that, the original super-PR came back. Then I tried some weird branching operation and I deleted the branch by mistake.

Long story short, yes I have to code on my computer still. I don't have time right now (I will tomorrow), but yeah, I'd love some help doing it the right way. I really want to learn how to 'git'.

I did download tortoise git, and I also downloaded git extensions, but I still ended up back in the git shell. I was using git extensions earlier when I killed the branch.

jdom commented:
Hehe,  yeah,  git tends to be tricky until it clicks,  and especially since there's 1000 different ways to do each thing. 
Just let me know when you have time and we can go over it through Skype or something.

sergeybykov commented:
@Joshua-Ferguson Non-hardcore-git-users of us are having pretty good experience with SourceTree. Interactive rebase and squashing are a pleasure to do there.

But yeah, there are too many ways to skin the git cat, fortunately and unfortunately at the same time. We all went through several months of a learning curve before getting comfortable with it.

Joshua-Ferguson commented:
@sergeybykov  I have installed SourceTree (it'd definitely got the best UI of the 3 (tortoise and git extensions)) @jdom I am available the rest of the day.

jdom commented:
Hi @Joshua-Ferguson, I reached out to you on Gitter, but in case you didn't see it: do you want to go over this now? jump in on the private chat in Gitter so that we set up a call

dnfclas commented:
Hi __@Joshua-Ferguson__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by .NET Foundation and real humans are currently evaluating your PR.</p>

TTYL, DNFBOT;

jdom commented:
Thanks a lot @Joshua-Ferguson, it took some time to battle with the tooling, but you won :)

Joshua-Ferguson commented:
Thanks! lol... I will always be looking for '.git 's now.
